gef-ui/server/routes/application-details.js

```diff
@@ -1,10 +1,11 @@
 const express = require('express');
 const { applicationDetails, postApplicationDetails } = require('../controllers/application-details');
-const { validateToken, validateBank } = require('../middleware');
+const { validateToken, validateBank, validateRole } = require('../middleware');
+const { MAKER, CHECKER, READ_ONLY } = require('../constants/roles');
 
 const router = express.Router();
 
-router.get('/application-details/:dealId', [validateToken, validateBank], (req, res) => applicationDetails(req, res));
-router.post('/application-details/:dealId', [validateToken, validateBank], (req, res) => postApplicationDetails(req, res));
+router.get('/application-details/:dealId', [validateToken, validateRole({ role: [MAKER, CHECKER, READ_ONLY] }), validateBank], (req, res) => applicationDetails(req, res));
+router.post('/application-details/:dealId', [validateToken, validateRole({ role: [MAKER, CHECKER] }), validateBank], (req, res) => postApplicationDetails(req, res));
```

Comment:
Why does `CHECKER` have permission here?

Reply:
As above

Reply:
I'm not convinced by this either - `postApplicationDetails` redirects to `/gef/application-details/${dealId}/submit` and only makers have access to `GET /gef/application-details/${dealId}/submit`

Reply:
updated